wooey/backend/utils.py

```diff
@@ -334,10 +340,17 @@ def mkdirs(path):
             raise
 
 
+def get_upload_path(filepath, checksum=None):
+    filename = os.path.split(filepath)[1]
+    if checksum is None:
+        checksum = get_checksum(filepath)
+    return os.path.join(wooey_settings.WOOEY_FILE_DIR, checksum, filename)
```

Comment:
This will give us one folder with all the hashes in it - large number of entries in a folder can cause performance issues. Maybe create a level of subfolders above e.g. `return os.path.join(wooey_settings.WOOEY_FILE_DIR, checksum[:2], checksum[2:], filename)` (.git apparently does something similar internally https://git-scm.com/book/en/v2/Git-Internals-Git-Objects ).